From 4a4aa43a57909d85fc204b052ef226dec0726283 Mon Sep 17 00:00:00 2001
From: Lukas Czerner <lczerner@redhat.com>
Date: Thu, 23 Oct 2014 09:54:25 -0400
Subject: [fs] ext4: address a benign compiler warning

Message-id: <1414058125-4183-112-git-send-email-lczerner@redhat.com>
Patchwork-id: 98222
O-Subject: [RHEL 7.1 PATCH v2 111/171] ext4: address a benign compiler warning
Bugzilla: 1150139
RH-Acked-by: Eric Sandeen <sandeen@redhat.com>

From: Patrick Palka <patrick@parcs.ath.cx>

BZ 1150139
https://bugzilla.redhat.com/show_bug.cgi?id=1150139

BREW 8142710
https://brewweb.devel.redhat.com/taskinfo?taskID=8142710

Upstream commit 024949ec8fc165bfac8eb051e537bc303adb365f

 When !defined(CONFIG_EXT4_DEBUG), mb_debug() should be defined as a
 no_printk() statement instead of an empty statement in order to suppress
 the following compiler warning:

 fs/ext4/mballoc.c: In function 'ext4_mb_cleanup_pa':
 fs/ext4/mballoc.c:2659:47: warning: suggest braces around empty body in an 'if' statement [-Wempty-body]
    mb_debug(1, "mballoc: %u PAs left\n", count);

 Signed-off-by: Patrick Palka <patrick@parcs.ath.cx>
 Signed-off-by: "Theodore Ts'o" <tytso@mit.edu>
 Signed-off-by: Lukas Czerner <lczerner@redhat.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/ext4/mballoc.h b/fs/ext4/mballoc.h
index 08481ee..9347328 100644
--- a/fs/ext4/mballoc.h
+++ b/fs/ext4/mballoc.h
@@ -48,7 +48,7 @@ extern ushort ext4_mballoc_debug;
   }       \
  } while (0)
 #else
-#define mb_debug(n, fmt, a...)
+#define mb_debug(n, fmt, a...)  no_printk(fmt, ## a)
 #endif
 
 #define EXT4_MB_HISTORY_ALLOC  1 /* allocation */
-- 
1.7.1